tensorflow/compiler/xla/service/gpu/gpu_compiler.cc

```diff
@@ -628,7 +629,11 @@ Status GpuCompiler::OptimizeHloModule(
   if (stream_exec != nullptr) {
     gpu_version = GetGpuVersion(stream_exec);
     se::dnn::DnnSupport* dnn = stream_exec->AsDnn();
-    TF_RET_CHECK(dnn != nullptr);
+    if (dnn == nullptr) {
+      return tsl::errors::FailedPrecondition(
+          "Not able to initialize the dnn library."
```

Comment:
Perhaps: "DNN library initialization failed. Look at the errors above for more details."

Ideally we would also plumb a better error out of the code above, but this might work for now.

Reply:
I amended the commit.
Agreed, doing proper error handling would be better.